### ISSUE TITLE
Version control

### ISSUE THREAD

**Issue by favoyang** [popcron/gizmos#20]
Hi @popcron,

Thanks for making this awesome UPM package. I'm the creator of the OpenUPM platform, an open-source UPM registry, and continuous build service based on Git tags. It's hosting more than 200+ open-source UPM packages at the time of writing.

To make it happen, I would suggest you make GitHub releases (which create Git tags) for version control. It can be achieved either manually or using [an automatic approach](https://medium.com/openupm/how-to-maintain-upm-package-part-2-f352fbf5f87c).

In your case, the new release would be `1.6.6` or `v1.6.6`. After your GitHub release, our build pipelines will detect the changes and build UPM packages in 5-10 minutes. Then you can visit your package at https://openupm.com/packages/com.popcron.gizmos/.

Hopeful this makes sense for you.

**Comment by popcron** [comment]
At the moment I have no interest in making this automatically available for the open upm platform as this is a package that still has some minor issues relating to the differences between the legacy render pipeline and SRP based ones. I'm also waiting to see what UT's approach to community packages will be in the coming months.

But I'll consider this anyways, thanks for bringing it up.

**Comment by favoyang** [comment]
Thanks for the quick repsonse. Do you have a source of the "UT's approach to community packages"? I'd like to follow-up.

I will close this issue for now. But feel free to re-open it if you'd like to give it a try.

**Comment by popcron** [comment]
There is nothing concrete for that kind of info, but they are working on features that make the idea of a registry possible, which is great for private projects: [link](https://forum.unity.com/threads/setup-for-scoped-registries-private-registries.573934/) that they recently added proper support for.

Basically, I'd rather wait and see if UT is interested in making plans that allow for a public and built-in registry for people to submit to. Though with the asset store being migrated to packages, it already seems like exactly what I'm talking about. It's built-in, people can submit to it and people can add an asset store package to their project through the pacman interface.

**Comment by favoyang** [comment]
Hi @popcron,

Thanks for the link. I've noticed they've added the support to auth, and a hacky way to work with GitHub registry. Both make sense for privacy registries.

For other parts, I shared a similar feeling. The reasons I create OpenUPM are
- It seems the Unity asset store is working well. It has many thousands of asset here, and will eventually support UPM format as the first citizen for both free and paid assets. There's no reason to stop using it.
- I want to fill the missing piece of the puzzle to the Unity eco-system, which is supporting and growing open-source projects in an open way. The process should be as low-friction as possible. By simply creating a Git tag and perhaps using a CI to automate.

I can not stop other forks to submit your packages, it basically how the system works, so I will leave it as now. The package has a pending state which will alert users it's not available yet. And they can always navigate to your website to install it via Git URL.
